framework/behaviors/SluggableBehavior.php

```diff
@@ -142,9 +148,12 @@ protected function getValue($event)
             if ($this->isNewSlugNeeded()) {
                 $slugParts = [];
                 foreach ((array) $this->attribute as $attribute) {
-                    $slugParts[] = ArrayHelper::getValue($this->owner, $attribute);
+                    $part = ArrayHelper::getValue($this->owner, $attribute);
+                    if ($this->skipOnEmpty && $this->isEmpty($part)) {
+                        return $this->owner->{$this->slugAttribute};
```

Comment:
I'd expect that. Your first point is how it behaves now. Usually if you're using a field to generate a slug the field is required.  Second point is what this property was introduced for.